NullPointerException in `DockerClientResolver` when running multiple Gradle `BuildImageTask`s in parallel

**Environment**:

- *Jib version:* 3.3.1
- *Build tool:* Gradle 8.0.1
- *OS:* Linux

**Description of the issue**:

We have multiple projects in our Gradle build which have the jib plugin applied to them and we are using [parallel execution](https://docs.gradle.org/current/userguide/multi_project_configuration_and_execution.html#sec:parallel_execution). This means that multiple instances `com.google.cloud.tools.jib.gradle.BuildImageTask` can execute in parallel. When that happens we are sometimes seeing the following exception (stacktrace truncated to significant parts only):
```
  Caused by: java.lang.NullPointerException: Cannot invoke "java.lang.ClassLoader.getParent()" because "this.currentLoader" is null
    at com.google.cloud.tools.jib.docker.DockerClientResolver.resolve(DockerClientResolver.java:38)
    at com.google.cloud.tools.jib.api.JibContainerBuilder.<init>(JibContainerBuilder.java:104)
    at com.google.cloud.tools.jib.api.Jib.from(Jib.java:87)
    at com.google.cloud.tools.jib.api.JavaContainerBuilder.from(JavaContainerBuilder.java:182)
    at com.google.cloud.tools.jib.plugins.common.PluginConfigurationProcessor.getJavaContainerBuilderWithBaseImage(PluginConfigurationProcessor.java:543)
    at com.google.cloud.tools.jib.plugins.common.PluginConfigurationProcessor.processCommonConfiguration(PluginConfigurationProcessor.java:425)
    at com.google.cloud.tools.jib.plugins.common.PluginConfigurationProcessor.processCommonConfiguration(PluginConfigurationProcessor.java:490)
    at com.google.cloud.tools.jib.plugins.common.PluginConfigurationProcessor.createJibBuildRunnerForRegistryImage(PluginConfigurationProcessor.java:309)
    at com.google.cloud.tools.jib.gradle.BuildImageTask.buildImage(BuildImageTask.java:116)
```

**Expected behavior**:
No NPEs being thrown when multiple instances `com.google.cloud.tools.jib.gradle.BuildImageTask` execute in parallel.

**Steps to reproduce**:
Run multiple instances of `com.google.cloud.tools.jib.gradle.BuildImageTask` in parallel. This is a concurrency issue so it's not always reproducible.

**Additional Information**: 
I think that `com.google.cloud.tools.jib.docker.DockerClientResolver#resolve` is not thread safe yet it can be called from multiple threads by the Gradle plugin when running with parallel execution enabled. It uses a reference to a static instance of `ServiceLoader` which according to its javadoc is not thread safe:

> Instances of this class are not safe for use by multiple concurrent threads.

Another stacktrace with a slightly different error caused by what I believe is the same issue:
```
    Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 2 out of bounds for length 2
12:56:37       at com.google.cloud.tools.jib.docker.DockerClientResolver.resolve(DockerClientResolver.java:38)
12:56:37       at com.google.cloud.tools.jib.api.JibContainerBuilder.<init>(JibContainerBuilder.java:104)
12:56:37       at com.google.cloud.tools.jib.api.Jib.from(Jib.java:87)
12:56:37       at com.google.cloud.tools.jib.api.JavaContainerBuilder.from(JavaContainerBuilder.java:182)
12:56:37       at com.google.cloud.tools.jib.plugins.common.PluginConfigurationProcessor.getJavaContainerBuilderWithBaseImage(PluginConfigurationProcessor.java:543)
12:56:37       at com.google.cloud.tools.jib.plugins.common.PluginConfigurationProcessor.processCommonConfiguration(PluginConfigurationProcessor.java:425)
12:56:37       at com.google.cloud.tools.jib.plugins.common.PluginConfigurationProcessor.processCommonConfiguration(PluginConfigurationProcessor.java:490)
12:56:37       at com.google.cloud.tools.jib.plugins.common.PluginConfigurationProcessor.createJibBuildRunnerForRegistryImage(PluginConfigurationProcessor.java:309)
12:56:37       at com.google.cloud.tools.jib.gradle.BuildImageTask.buildImage(BuildImageTask.java:116)
```

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/docker/DockerClientResolver.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/docker/DockerClientResolver.java
@@ -23,9 +23,6 @@ import java.util.ServiceLoader;
 
 public class DockerClientResolver {
 
-  private static final ServiceLoader<DockerClient> dockerClients =
-      ServiceLoader.load(DockerClient.class);
-
   private DockerClientResolver() {}
 
   /**
@@ -35,6 +32,7 @@ public class DockerClientResolver {
    * @return dockerClient if any is found
    */
   public static Optional<DockerClient> resolve(Map<String, String> parameters) {
+    ServiceLoader<DockerClient> dockerClients = ServiceLoader.load(DockerClient.class);
     for (DockerClient dockerClient : dockerClients) {
       if (dockerClient.supported(parameters)) {
         return Optional.of(dockerClient);

